asdf_astropy/converters/coordinates/earth_location.py

```diff
@@ -3,10 +3,15 @@
 
 class EarthLocationConverter(Converter):
     tags = ("tag:astropy.org:astropy/coordinates/earthlocation-*",)
-    types = ("astropy.coordinates.earth.EarthLocation",)
+    types = (
+        "astropy.coordinates.earth.EarthLocation",
+        "astropy.utils.masked.core.MaskedEarthLocation",
+    )
 
     def to_yaml_tree(self, obj, tag, ctx):
-        return obj.info._represent_as_dict()
+        result = obj.info._represent_as_dict()
+        result.pop("__class__", None)
```

Comment:
Under what condition was `__class__` being included in the `_represent_as_dict` result?

Reply:
```pycon
>>> from astropy import units as u
>>> from astropy.coordinates import EarthLocation
>>> from astropy.utils.masked import Masked
>>> lon = Masked([0, 1] * u.deg)
>>> lat = Masked([2, 3] * u.deg)
>>> el = EarthLocation.from_geodetic(lon, lat)
>>> el.info._represent_as_dict()
{'x': <MaskedQuantity [6374277.59793627, 6368484.28240152] m>, 'y': <MaskedQuantity [     0.        , 111162.30664365] m>, 'z': <MaskedQuantity [221104.54531112, 331574.31534289] m>, 'ellipsoid': 'WGS84', '__class__': 'astropy.coordinates.earth.EarthLocation'}
```

Reply:
@taldcroft, @mhvk, I most often see your handles associated with `_represent_as_dict` methods. Is it a bug for the `__class__` key to be present in its return value for masked classes?